### ISSUE TITLE
Cannot invoke "java.util.List.stream()" because the return value of "java.util.Map.get(Object)" is null

### ISSUE THREAD

**Issue by rjbgaspar** [Playtika/feign-reactive#413]
Hi,

First of all, congratulations on your work.

I'm trying to run a request and I'm getting the following error:
```
2021-11-24 09:12:56.663  INFO 56828 --- [           main] com.example.demo.DemoApplication         : Starting DemoApplication using Java 17 on VAL011 with PID 56828 (C:\Users\Ricardo Gaspar\code\labs\feign-reactive-error\target\classes started by Ricardo Gaspar in C:\Users\Ricardo Gaspar\code\labs\feign-reactive-error)
2021-11-24 09:12:56.667  INFO 56828 --- [           main] com.example.demo.DemoApplication         : No active profile set, falling back to default profiles: default
2021-11-24 09:12:58.875  INFO 56828 --- [           main] o.s.b.web.embedded.netty.NettyWebServer  : Netty started on port 8080
2021-11-24 09:12:58.893  INFO 56828 --- [           main] com.example.demo.DemoApplication         : Started DemoApplication in 2.867 seconds (JVM running for 5.006)
2021-11-24 09:12:59.269 ERROR 56828 --- [           main] reactor.core.publisher.Operators         : Operator called default onErrorDropped

reactor.core.Exceptions$ErrorCallbackNotImplemented: java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "java.util.Map.get(Object)" is null
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because the return value of "java.util.Map.get(Object)" is null
	at reactivefeign.methodhandler.PublisherClientMethodHandler.lambda$queries$6(PublisherClientMethodHandler.java:172) ~[feign-reactor-core-3.1.1.jar:na]
	at java.base/java.util.HashMap$KeySet.forEach(HashMap.java:1007) ~[na:na]
	at reactivefeign.methodhandler.PublisherClientMethodHandler.queries(PublisherClientMethodHandler.java:171) ~[feign-reactor-core-3.1.1.jar:na]
	at reactivefeign.methodhandler.PublisherClientMethodHandler.buildUri(PublisherClientMethodHandler.java:122) ~[feign-reactor-core-3.1.1.jar:na]
	at reactivefeign.methodhandler.PublisherClientMethodHandler.buildRequest(PublisherClientMethodHandler.java:107) ~[feign-reactor-core-3.1.1.jar:na]
	at reactivefeign.methodhandler.PublisherClientMethodHandler.invoke(PublisherClientMethodHandler.java:98) ~[feign-reactor-core-3.1.1.jar:na]
	at reactivefeign.methodhandler.PublisherClientMethodHandler.invoke(PublisherClientMethodHandler.java:51) ~[feign-reactor-core-3.1.1.jar:na]
	at reactivefeign.methodhandler.MonoMethodHandler.invoke(MonoMethodHandler.java:18) ~[feign-reactor-core-3.1.1.jar:na]
	at reactivefeign.methodhandler.MonoMethodHandler.invoke(MonoMethodHandler.java:6) ~[feign-reactor-core-3.1.1.jar:na]
	at reactivefeign.ReactiveInvocationHandler.invoke(ReactiveInvocationHandler.java:64) ~[feign-reactor-core-3.1.1.jar:na]
	at com.example.demo.$Proxy50.queryForList(Unknown Source) ~[na:na]
	at com.example.demo.DemoApplication.lambda$runner$2(DemoApplication.java:37) ~[classes/:na]
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:758) ~[spring-boot-2.6.0.jar:2.6.0]
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:748) ~[spring-boot-2.6.0.jar:2.6.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:309) ~[spring-boot-2.6.0.jar:2.6.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1301) ~[spring-boot-2.6.0.jar:2.6.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1290) ~[spring-boot-2.6.0.jar:2.6.0]
	at com.example.demo.DemoApplication.main(DemoApplication.java:22) ~[classes/:na]

------------- GET queryForList with FeignClient completed ------------------
```

I don't understand why, however the API I'm querying is std MVC app

This is the code:
```java
@SpringBootApplication
public class DemoApplication {

	public static void main(String[] args) {
		SpringApplication.run(DemoApplication.class, args);
	}

	@Bean
	ApplicationRunner runner() {
		return args -> {
			/* Create instance of your API */
			ErpSQLQueryApi client =
					WebReactiveFeign
							.<ErpSQLQueryApi>builder()
							.target(ErpSQLQueryApi.class, "https://...");

			client.queryForList()
					.doOnNext(it -> System.out.println(it.getData()))
					.doFinally(s -> System.out.println("------------- GET queryForList with FeignClient completed ------------------"))
					.then()
					.subscribe();
		};
	}
}

@Headers({"Accept: application/json"})
interface ErpSQLQueryApi {

	@RequestLine("GET /api/v1/query/erp-sage-x3-acl-247-starter-processor?manufacturingPlan=ZPH-V121-00123&workOrder=")
	Mono<ACL247StarterProcessDTO> queryForList();
}

@Data
@NoArgsConstructor
class ACL247StarterProcessDTO {
	private List<Map<String, Object>> data;
}
```
- If i use WebClient to make the request everything works well.

Thanks in advance

**Comment by kptfh** [comment]
Hi. Hadn't tested it with Java 17 yet.
Can you check it with Java 11?

**Comment by kptfh** [comment]
Please fork this repository and add test so I can debug it.

**Comment by kptfh** [comment]
managed to reproduce this issue.
the root cause is this request parameter: `&workOrder=`

**Comment by rjbgaspar** [comment]
Hello,
Sorry for the late reply.

This HTTP request maps the parameters to an SQL query where both parameters are used. 

The lack on a parameter means all result should be return. 

Something like this

```sql
SELECT
    *
FROM ...
WHERE
    MFI.TSICOD_0 = 'P'
AND (
            (:manufacturingPlan = '' OR SOQ.ZMFGPLN_0 = :manufacturingPlan)
        AND (COALESCE(:workOrder,'') = '' OR MFI.MFGNUM_0 IN (:workOrder))
)
```
I couldn't find another way to make it work. 

Don't you think that this should be addressed?

Thanks in advance

**Comment by kptfh** [comment]
fix released in 3.1.4
